### ISSUE TITLE
Sporadic failure of parallel.py test

### ISSUE THREAD

**Issue by awemany** [gandrewstone/bitcoin#566]
As seen for this travis-ci build:

https://travis-ci.org/BitcoinUnlimited/BitcoinUnlimited/builds/231253518

Comments from slack on this:
```
<ptschip> oh i see i happened only once...it's proabably the known bug that happens sometimes during a re-org..haven't had time to fix it with all that is going on right now.
<freetrader> yes, it was directly after a reorg test step
<freetrader> i think it was a 'reorg 1' or sth
<freetrader> block chain heights not as expected AFAIR
<freetrader> (failed an assertion)
<ptschip> ok...yeah it's hard to trigger...basically if we're in the middle of reorg and we receive another block it can cause the chain not to catch up until another block is mined.
```

Relevant extract from travis:

```
<thezerg> I already restarted but here's the log:
<thezerg> check for re-org 1
<thezerg> Assertion failed: 7b3e25fddeed6afe8479d234a63a4fc2c45464238c06da93ad0ba1c8f834294c != 77e90873927f72875e01d3d422bbc09e7e9ec393cfc1989ca673df41e1487dc1
<thezerg>   File "/home/travis/build/BitcoinUnlimited/BitcoinUnlimited/qa/rpc-tests/test_framework/test_framework.py", line 184, in main
<thezerg>     self.run_test()
<thezerg>   File "/home/travis/build/BitcoinUnlimited/BitcoinUnlimited/build/../qa/rpc-tests/parallel.py", line 249, in run_test
<thezerg>     assert_equal(self.nodes[1].getbestblockhash(), self.nodes[2].getbestblockhash())
<thezerg>   File "/home/travis/build/BitcoinUnlimited/BitcoinUnlimited/qa/rpc-tests/test_framework/util.py", line 612, in assert_equal
<thezerg>     raise AssertionError("%s != %s"%(str(thing1),str(thing2)))
<thezerg> Stopping nodes
<thezerg> Cleaning up
```

**Comment by sickpig** [comment]
@awemany wonder if recent work by pstchip in this area have fixed the problem. Can't see any `parrallel.py` failures recently on travis history.

**Comment by ptschip** [comment]
fixed in #791

**Comment by awemany** [comment]
Ok, closing this then (didn't do any long term tests whether no more parallel.py of that kind occur).
